### ISSUE TITLE
Failures in ParentChildClientYamlTestSuiteIT

### ISSUE THREAD

**Issue by danhermann** [elastic/elasticsearch#70270]
There are failures in three tests in this test suite. Excerpts and repro lines for one of them are below, but all are available in the linked build scans.

**Build scan**:
https://gradle-enterprise.elastic.co/s/5flx6k6jaabhu
https://gradle-enterprise.elastic.co/s/2y4jh76yjt3ti

**Repro line**:
`./gradlew ':modules:parent-join:yamlRestTest' --tests "org.elasticsearch.join.ParentChildClientYamlTestSuiteIT.test {yaml=/50_order_by/order by sub agg containing join}" -Dtests.seed=5DFF4DD8A9ED1A44 -Dtests.security.manager=true -Dtests.locale=nl-NL -Dtests.timezone=Etc/GMT+8 -Druntime.java=8	
`

**Reproduces locally?**: Yes

**Applicable branches**: 7.11

**Failure history**: Appears to have started on 3/10/21 with the release of 7.11.2

**Failure excerpt**:

```
java.lang.AssertionError: Failure at [/50_order_by:68]: aggregations.name.buckets.0.key didn't match expected value: |  
-- | --
  | aggregations.name.buckets.0.key: expected String [pig] but was String [cow]
```

**Comment by nik9000** [comment]
@imotov this sounds like my mess that you recently cleaned up.

**Comment by imotov** [comment]
Which probably means it's my mess now :)

**Comment by martijnvg** [comment]
Another instance of these test failures on the 7.11 branch: https://gradle-enterprise.elastic.co/s/6xvotp3vowas2

**Comment by andreidan** [comment]
Another instance on a 7.11 PR https://gradle-enterprise.elastic.co/s/p5afwb23xcxxo

**Comment by nik9000** [comment]
I'll grab this when I've got some coffee.

**Comment by nik9000** [comment]
OK! I've muted. The weren't actually running the test until we released 7.11.2 because we skipped every version up to and including 7.11.2. So now that we bumped the 7.11 branch to 7.11.3-SNAPSHOT its running. And, I guess, the fix didn't work in that branch.

**Comment by nik9000** [comment]
I'll consult with @imotov when he has a chance about what we should do with 7.11. ^^^^ is why the backport worked but started failing after 7.11. I don't know why the backport didn't make the test pass but I'm not sure its worth figuring out because 7.12 is already forked.

**Comment by nik9000** [comment]
We should probably update the skip version in the newer branches. It doesn't fail because we don't run bwc tests on those yml test.

**Comment by imotov** [comment]
OK, that one was a head scratcher. Turned out that I fat fingered the backport and enabled these tests instead of keeping them muted as it is done in other branches. These tests still have to be fixed, but it is tracked by #66876, so I am going to fix the skip message and close this issue. Sorry for all the confusion.

**Comment by imotov** [comment]
Fixed by ca9eebbb30dad595a808419ca250676ad8c1b901
